### PR TITLE
SentinelOne: PAPP-29252 Update on-poll action to use container_count

### PR DESCRIPTION
**Pull request by gary-phantom**
Please ensure your pull request (PR) adheres to the following guidelines:

- Please refer to our contributing documentation for any questions on submitting a pull request, link: [Contribution Guide](https://github.com/splunk-soar-connectors/.github/blob/main/.github/CONTRIBUTING.md)

This PR fixes the hardcoded max_container creation count of 1000 to potentially alleviate the memory exhaustion. 
Instead of the hardcoded limit, the fix uses container_count action parameter in the on-poll action. 

**Due to the patch nature of the fix, debug_print & min_phantom_version checks will be overriden.**

#### Please check if your PR fulfills the following requirements:
- [ ] Testing of all the changes has been performed (for bug fixes / features)
- [ ] The readme.html has been reviewed and added / updated if needed (for bug fixes / features)
- [x] Use the following format for the PR description: `<App Name>: <PR Type> - <PR Description>`
- [x] Provide release notes as part of the PR submission which describe high level points about the changes for the upcoming GA release.
- [ ] Verify all checks are passing.
- [ ] Do NOT use the `next` branch of the forked repo. Create separate feature branch for raising the PR.
- [ ] Do NOT submit updates to dependencies unless it fixes an issue.
#### Please check the type of change your PR introduces:
- [ ] New App
- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Documentation
- [ ] Other (please describe): 

## Security Considerations (REQUIRED)
- [ ] If you are exposing any endpoints using a [REST handler](https://docs.splunk.com/Documentation/SOAR/current/DevelopApps/RESTHandlers), 
  please document them in the `readme.html`.
- [ ] If this is a new connector or you are adding new actions
    - [ ] Please document in the `readme.html` all methods (eg, OAuth) used to authenticate 
      with the service that the connector is integrating with.
    - [ ] If any actions are unable to run on SOAR Cloud, please document this in the `readme.html`.
- [ ] Are you introducing any new cryptography modules? If yes, please elaborate their purpose:
- [ ] Are you are accessing the file system? If yes, please verify that you are only accessing paths returned through 
the [Vault](https://docs.splunk.com/Documentation/SOARonprem/5.2.1/DevelopApps/AppDevAPIRef#Vault) API.
- [ ] Are you are marking code to be ignored by Semgrep with [`nosemgrep`](https://semgrep.dev/docs/ignoring-files-folders-code/#ignoring-code-through-nosemgrep)?
If yes, please provide justification in an additional comment next to the ignored code.

  

## Release Notes (REQUIRED)
- Provide release notes as part of the PR submission which describe high level points about the changes for the upcoming GA release.

## What is the current behavior? (OPTIONAL)
- Describe the current behavior that you are modifying.

## What is the new behavior? (OPTIONAL)
- Describe the behavior or changes that are being added by this PR.

## Other information (OPTIONAL)
- Any other information that is important to this PR such as screenshots of how the component looks before and after the change.

## Pay close attention to (OPTIONAL)
- Any specific code change or test case points which must be addressed/reviewed at the time of GA release.

## Screenshots (if relevant)
This PR corrects the proper usage of container_count parameter in the on-poll action which was hardcoded to 1000 previously.  To maintain backward compatibility, 1000 is still used if the parameter is not used. 

Screen Shot 2023-01-22 at 4 25 57 PM
Screen Shot 2023-01-23 at 3 54 40 PM

---
Thanks for contributing!

### CODE REVIEW

File: sentinelone_connector.py [hunk 1]
```diff
@@ -1308,22 +1308,22 @@ def _handle_on_poll(self, param):
...
-                self.debug_print(('Error while creating artifacts for container with ID {container_id}. {error_msg}').format(
-                    container_id=container_id, error_msg=artifacts_creation_msg))
+                self.debug_print(('Error while creating artifacts for container with ID {container_id}. {error_message}').format(
+                    container_id=container_id, error_message=artifacts_creation_message))
         self._state['first_run'] = False
         self._state['last_ingestion_time'] = end_time
         return action_result.set_status(phantom.APP_SUCCESS)
 
-    def _get_alerts(self, action_result, start_time, end_time, max_limit=None):
+    def _get_alerts(self, action_result, start_time, end_time, max_limit=1000):
```
Comment (rgil-splunk) [inline sentinelone_connector.py]: should we set this default to something smaller? like 50 or 100?

**Review by rgil-splunk — APPROVED** [review]

File: sentinelone_connector.py (hunk 1 above)
Comment (gary-phantom) [reply sentinelone_connector.py]: I've considered that but felt that maintaining backward compatibility is better but I'm open to suggestion.

Comment (gary-phantom) [reply sentinelone_connector.py]: on 2nd thought, I do like the idea of using 100 as default. 
I could add that as a default value in the APP JSON definition as well. 
Would that be acceptable?

Comment (mpan-splunk) [reply sentinelone_connector.py]: Should we put this fixed number in sentinelone_consts.py ?

Comment (gary-phantom) [reply sentinelone_connector.py]: Agreed. 
I changed the limit to 100 and added the fix number in the const file. 
Also, updated APP JSON to flag unused on-poll action parameters.

**Review by mpan-splunk — COMMENTED** [review]
"min_phantom_version": "5.1.0", we can change this.

**Comment by gary-phantom** [comment]
> 

@mpan-splunk this is a patch release so min-phantom-version will remain as is.

**Review by mpan-splunk — APPROVED** [review]

**Pull request merged**